core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -974,6 +984,9 @@ public NativeImageInvokerInfo build() {
                     }
                 }
 
+                nativeImageAgentConfigDirectory
+                        .ifPresent(dir -> nativeImageArgs.add("-H:ConfigurationFileDirectories=" + dir.getDirectory()));
```

Comment:
This is overriding the default, but that's OK since we merge the native-image-agent configurations with the ones generated by Quarkus, so everything will end up under this directory, right?

Reply:
This was a very interesting comment. The bottom line is that the line I've added doesn't override the default. It complements it. IOW, any `META-INF/native-image/*.json` the native image build generates will carry through, as well as anything inside the `-H:ConfigurationFileDirectories` above.

You can see this in action (I will adjust the test to verify this) by doing the following. I added `Carol` to the native image agent IT in this PR that looks like this:

```java
@RegisterForReflection
public class Carol
{
    public String sayMyName()
    {
        return "Carol";
    }
}
```

Then I made sure that the unit test did not exercise this class, so the native image agent reflection configuration was:

```json
[
{
  "name":"org.acme.Alice",
  "methods":[{"name":"<init>","parameterTypes":[] }, {"name":"sayMyName","parameterTypes":[] }]
},
{
  "name":"org.acme.Bob"
}
]
```

Through annotation processing, `Carol` gets added to `META-INF/native-image/reflect-config.json`:

```
  {
    "name": "org.acme.Carol",
    "allDeclaredConstructors": true,
    "allDeclaredMethods": true,
    "allDeclaredFields": true
  },
```

Finally, I run the native executable and I was able to make this call successfully:

```
❯ curl http://localhost:8080/hello/Carol
Hello Carol%
```

This means that the registrations generated by Quarkus are complementary.